blogs/13_bfcl_v3_multi_turn.html

```diff
@@ -748,73 +751,59 @@ <h4>5. API Code Validation (🧑‍💻+ 💻 )</h4>
-            are designed to handle different function call patterns—simple, parallel, and nested—across multiple
-            rounds of interaction. The distinction between these call patterns is explained in the previous section.
+          <h3 id="inference">Model Inference & Execution</h3>
+          <ol>
+            <li>
+              Each entry comes with its own <code>initial_config</code>, which is used to initialize the API backend state. For example, a file system backend might start with a set of pre-existing files, and a messaging API might start with a specific message inbox history. The <code>initial_config</code> is loaded into the API backend at the beginning of each test entry, but it is not provided to the model directly, as the model can call exploration functions to figure out the state of the system and retrieve the necessary information.
+            </li>
+            <li>
+              At the beginning of each turn, a user query is provided to the model, along with the tools available for the task. A <code>step_counter</code> is also initialized to track the number of steps used in this turn; it resets to 0 at the beginning of each turn.
+            </li>
+            <li>
+              If the model makes any function calls (i.e., <code>decode_exec</code> in the model handler returns a non-empty list), the function calls are executed in the API backend (in order) and their execution results are provided back to the model. The current turn continues.
+            </li>
+            <li>
+              If the model doesn't output any valid function calls, we consider this the end of the current turn, and move on to the next turn. This could be due to either the model outputs chatting messages, or the model response cannot be properly decoded into executable function calls.
+            </li>
+            <li>
+              If the model takes more than 20 steps in a turn, the turn is force-terminated and this test entry is marked as wrong; no further turns will be queried. This usually happens when the model gets stuck in a loop or is unable to make progress (e.g., keep calling <code>ls</code> in the Gorilla File System entries).
+            </li>
+          </ol>
+          <h4>Note For Multi Turn Missing Function Category</h4>
+          <p>
+            In this category, one or more functions are held out from the function list provided to the model at the beginning; they will be provided in a later turns (say, turn 3). For FC models, the added functions will just be appended to the <code>tools</code> list. But for prompting models, since we supplied all the tools at the beginning in the system prompt and it's not possible to modify the system prompt in the middle of the conversation, we will provide the held-out function definitions in the content of a user message instead.
           </p>
```

Comment:
Conceptually, I want to see a side-by-side of how miss function is handled differently from other categories. 

A visual can help.

Reply:
Can add this item to backlog as well.